client/app/page.tsx

```diff
@@ -1,44 +1,46 @@
 "use client";
 import { useState, useEffect } from "react";
-import { ethers } from "ethers";
+import { ethers, BaseContract , BrowserProvider } from "ethers";
 import styles from "./page.module.css"
 import NFT1 from "../../artifacts/contracts/NFT.sol/NFT.json";
 import { toast, ToastContainer } from "react-toastify";
 import "react-toastify/dist/ReactToastify.css";
 import Image from "next/image";
 import nftImage from "./assets/images/nft.jpg";
-import NFT2 from "./components/nft";
-import { BrowserProvider } from "ethers";
-import { NFT__factory, NFT } from "../../typechain-types";
-import { BaseContract } from "ethers";
+import {NFT} from "../../typechain-types";
+
+
+
 
 
 export default function Home() {
   const [account, setAccount] = useState<string>("");
+  const [signer , setSigner]  = useState<ethers.JsonRpcSigner>();
   const [contract, setContract] = useState< NFT | null>(null);
   const [loading, setLoading] = useState<boolean>(false);
   const [burnloading, setBurnloading] = useState<boolean>(false);
   const [connect, setConnect] = useState<string>("");
 
   const [mintPrice, setMintPrice] = useState<string>("0");
-  const [totalSupply, setTotalSupply] = useState<number>(0);
+
   const [burnPrice, setBurnPrice] = useState<string>("0");
 
   const contractAddress = "0x210a37E01089d1160A9af5D32bDFAc7348FB99e6";
 
   useEffect(() => {
-    setConnect(localStorage.getItem("address") || "");
+    setConnect(localStorage.getItem("address") ?? "");
   }, []);
 
   useEffect(() => {
     const provider = new BrowserProvider(window.ethereum);
```

Comment:
define this globally. don't define this inside every function.

Reply:
Done